### PR TITLE
Make GlobalizationMode code consistent

### PR DESCRIPTION
**Pull request by tarekgh**
Address the comment https://github.com/dotnet/runtime/pull/54247#discussion_r661935281 in the PR https://github.com/dotnet/runtime/pull/54247

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @marek-safar

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs [hunk 1]
```diff
@@ -11,23 +11,17 @@ internal static partial class GlobalizationMode
         // split from GlobalizationMode so the whole class can be trimmed when Invariant=true.
         private static partial class Settings
         {
-            internal static bool Invariant { get; } = GetInvariantSwitchValue();
-            internal static readonly bool PredefinedCulturesOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
+            internal static bool Invariant { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
+            internal static bool PredefinedCulturesOnly { get; } =AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:15]:
```suggestion
            internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for the follow up.

**Review by safern — COMMENTED** [review]
LGTM

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs [hunk 2]
```diff
@@ -11,23 +11,17 @@ internal static partial class GlobalizationMode
         // split from GlobalizationMode so the whole class can be trimmed when Invariant=true.
         private static partial class Settings
         {
-            internal static bool Invariant { get; } = GetInvariantSwitchValue();
-            internal static readonly bool PredefinedCulturesOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
+            internal static bool Invariant { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
+            internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```
Comment (marek-safar) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:15]: nit: I think using read-only fields would be more appropriate for a private class

**Pull request merged**